### DISCUSSION TITLE
SubCategory isn't hidden when invisible is set to true

### DISCUSSION THREAD

**Discussion by spowell-216** [iTwin/itwinjs-core#5485] [Q&A] [answered]
I am creating an iModel which contains rooms.  I have set the default subcategory to invisible, but when rendering the iModel, the rooms are visible.  
room elements are created as `VolumeElement ` with valid geometry streams.  
The room category is being created as follows:
` categoryId = SpatialCategory.insert(iModelDb, IModelDb.dictionaryId, categoryName, new SubCategoryAppearance({ material: materialId, invisible: true}));`
and when i look in iModelConsole, i can see it is set
![image](https://user-images.githubusercontent.com/51784707/236325116-3603cab7-cd9f-45a0-bd3b-19cbd8c6dfbb.png)
but when I open the iModel, they are visible. 

I would like to keep these rooms hidden by default, but still have them available to do spatial queries to determine what is inside these rooms.  
Any help would be much appreciated.

**Comment by pmconne** [comment]
Please show the code that creates the `ViewState` to be displayed in the `Viewport`.

**Comment by spowell-216** [comment]
@pmconne I am not sure I follow, probably why I am having an issue.

after registering the categories, I register a default `CategorySelector`
`CategorySelector.insert(iModel, IModelDb.dictionaryId, description, categories)` 

then register the `DisplayStyle`
`DisplayStyle3d.insert(iModel, IModelDb.dictionaryId, styleName)`

then add all my spatial elements.

and before writing back to the hub, I create a `SpatialViewDefinition`
`SpatialViewDefinition.insertWithCamera(props.iModelDb, props.definitionModelId, props.viewName, props.modelSelectorId, props.spatialCategroySelectorId, props.displayStyleId, cameraRange, StandardViewIndex.Iso)`

**Comment by pmconne** [comment]
@spowell-216 if I understand you correctly when you say

> when I open the iModel, they are visible

it means you are opening some viewing application that display graphics for the iModel, and graphics belonging to your invisible subcategory are displayed unexpectedly.
If I misunderstand, please correct me.
Otherwise, please clarify what viewing application you are using and how you are going about opening and viewing the iModel in it.

**Reply by spowell-216** [reply]
@pmconne the viewer I am using is [iTwin Hub ](https://itwin.bentley.com/).

Basically what I have is one category, 'Room', and I have set it's default subcategory property to invisible = true, it has no other subcategories and when I view the iModel from the iTwin Hub, the geometry is visible.

**Reply by pmconne** [reply, marked as answer]
If by "view the iModel from the iTwin Hub", you mean, click a button to "View iModel" that opens a design review application, then your problem is that the design review application intentionally makes all categories and subcategories visible in the default view it opens.
You should be able to access the spatial view you created in your code via "Desktop Views", and observe that geometry belonging to your invisible subcategory does not display in that view.

**Reply by spowell-216** [reply]
@pmconne I think I understand now, the design review application is overriding my defaults.  I can create a new Desktop View hiding what i want and it does render correctly the next time its loaded.  thanks.  you can consider this closed.
